liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -0,0 +1,889 @@
+        log.info("Running goal: install-feature");
+        runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "install-feature", serverName, null);
+        log.info("Running goal: install-apps");
+        runMojo("net.wasdev.wlp.maven.plugins:liberty-maven-plugin", "install-apps", serverName, null);
+        
+        // resource directories
+        List<File> resourceDirs = new ArrayList<File>();
+        if (outputDirectory.exists()) {
+            List<Resource> resources = project.getResources();
+            for (Resource resource : resources) {
+                File resourceFile = new File(resource.getDirectory());
+                if (resourceFile.exists()) {
+                    resourceDirs.add(resourceFile);
+                }
+            }
+        }
+        if (resourceDirs.isEmpty()) {
+            File defaultResourceDir = new File(project.getBasedir() + "/src/main/resources");
+            log.debug("No resource directory detected, using default directory: " + defaultResourceDir);
+            resourceDirs.add(defaultResourceDir);
+        }
+
+        util = new DevMojoUtil(serverDirectory, sourceDirectory, testSourceDirectory, configDirectory, resourceDirs);
+        
+        util.addShutdownHook(executor);
+
+        util.enableServerDebug(libertyDebugPort);
+
+        util.startServer();
+
+        // collect artifacts canonical paths in order to build classpath
+        List<String> artifactPaths = new ArrayList<String>();
+        util.getArtifacts(artifactPaths);
+
+        if (hotTests && testSourceDirectory.exists()) {
+            // if hot testing, run tests on startup and then watch for keypresses
+            util.runTestThread(false, executor, -1, false, false);
+        } else {
+            // else watch for keypresses immediately
+            util.runHotkeyReaderThread(executor);
+        }
+
+        // pom.xml
+        File pom = project.getFile();
+        
+        util.watchFiles(pom, outputDirectory, testOutputDirectory, executor, artifactPaths, configFile);
+    }
+
+    private void addArtifacts(org.eclipse.aether.graph.DependencyNode root, List<File> artifacts) {
+        if (root.getArtifact() != null) {
+            artifacts.add(root.getArtifact().getFile());
+        }
+
+        for (org.eclipse.aether.graph.DependencyNode node : root.getChildren()) {
+            addArtifacts(node, artifacts);
+        }
+    }
+
+    private List<Artifact> getNewDependencies(List<Dependency> dependencies, List<Dependency> existingDependencies)
+            throws MojoExecutionException {
+        List<Artifact> updatedArtifacts = new ArrayList<Artifact>();
+        for (Dependency dep : dependencies) {
+            boolean newDependency = true;
+            for (Dependency existingDep : existingDependencies) {
+                if (dep.getArtifactId().equals(existingDep.getArtifactId())) {
+                    newDependency = false;
```

Comment:
You can break out of the inner for loop once you set newDependency to false.

Reply:
Added break statement